pkg/fedresolver/fedschema.go

```diff
@@ -0,0 +1,175 @@
+package fedresolver
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+// Define the httpClient interface
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// clientWrapper is a generic wrapper around http.Client or any httpClient implementation
+type clientWrapper struct {
+	client httpClient
+}
+
+func (c *clientWrapper) Do(req *http.Request) (*http.Response, error) {
+	return c.client.Do(req)
+}
+
+// Your SearchSchema struct with an interface for the HTTP client
+type SearchSchema struct {
+	routeToken []map[string]string
+	client     httpClient
+}
+
+type schemaPayload struct {
+	Data struct {
+		SearchSchema map[string][]string `json:"searchSchema"`
+	} `json:"data"`
+}
+
+func SearchSchemaResolver(ctx context.Context) (map[string]interface{}, error) {
+	routeToken := getTokenAndRoute()
+
+	// Create an http.Client
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+		Timeout: 30 * time.Second,
+	}
+
+	// Wrap the http.Client with the clientWrapper
+	wrappedClient := &clientWrapper{client: httpClient}
+
+	// Create a SearchSchema instance with the wrapped client
+	searchSchemaResult := &SearchSchema{
+		routeToken: routeToken,
+		client:     wrappedClient,
+	}
+
+	return searchSchemaResult.searchSchemaResults(ctx)
+}
+
+func (s *SearchSchema) searchSchemaResults(ctx context.Context) (map[string]interface{}, error) {
+	klog.Info("Resolving searchSchemaResults()")
+	srchSchema := map[string]interface{}{}
+	var schemaErr error
```

Comment:
This should be an array to capture if there's multiple errors from different targets.
```suggestion
	var schemaErr [] error
```